### PR TITLE
[Lib][Add] googletest

### PR DESCRIPTION
**Pull request by notvoveex**
Changes:
- add: submodule lib/googletest

### CODE REVIEW

**Comment by lucky-rydar** [comment]
Add means integrate with cmake project to be ready to use

**Comment by lucky-rydar** [comment]
It is better to add simple test for main file such "main_test.cpp" to show that tests work correctly

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,10 @@
...
+project(DATALINE)
 
 set(CMAKE_CXX_STANDARD 17)
 
-add_executable(mydb src/main.cpp)
+include_directories(src)
+
+add_subdirectory(lib/googletest)  # GTEST_DIR
+
+add_executable(DATALINE src/main.cpp)
```
Comment (lucky-rydar) [inline CMakeLists.txt]: here update to link all the .cpp and/or .h files

**Comment by Vov11x** [comment]
> Add means integrate with cmake project to be ready to use

The tests are ready to be used right now.

**Comment by Vov11x** [comment]
> It is better to add simple test for main file such "main_test.cpp" to show that tests work correctly

Added

File: CMakeLists.txt (hunk 1 above)
Comment (Vov11x) [reply CMakeLists.txt]: > here update to link all the .cpp and/or .h files

Already updated.

**Comment by notvoveex** [comment]
> Add means integrate with cmake project to be ready to use

The tests are ready to be used right now.



> It is better to add simple test for main file such "main_test.cpp" to show that tests work correctly

Added.

File: CMakeLists.txt (hunk 1 above)
Comment (notvoveex) [reply CMakeLists.txt]: > here update to link all the .cpp and/or .h files

Already updated.

**Pull request merged**